File: src/types/domain/request/merchant/list.ts
export interface FindAllMerchant {
  search: string;
  page: number;
  page_size: number;
  toast: any;
}

export interface FindAllMerchantTransaction {
  merchant_id: number;
  search: string;
  page: number;
  page_size: number;
  toast: any;
}

export interface FindAllTransactionByApiKey {
  api_key: string;
  search: string;
  page: number;
  page_size: number;
  toast: any;
}

File: src-tauri/src/domain/requests/merchant/list.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct FindAllMerchant {
    pub search: String,
    pub page: u32,
    pub page_size: u32,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct FindAllMerchantTransaction {
    pub merchant_id: u32,
    pub search: String,
    pub page: u32,
    pub page_size: u32,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct FindAllMerchantTransactionApiKey {
    pub api_key: String,
    pub search: String,
    pub page: u32,
    pub page_size: u32,
}
